tests/unit_tests/corsika/test_corsika_output.py

```diff
@@ -1,29 +1,750 @@
+            in caplog.text
+        )
+    assert hist_config == corsika_output_instance._create_histogram_default_config()
+
+
+def test_hist_config_save_and_read_yml(corsika_output_instance, io_handler):
+    # Test producing the yaml file
+    temp_hist_config = corsika_output_instance.hist_config
+    corsika_output_instance.hist_config_to_yaml()
+    output_file = io_handler.get_output_file(file_name="hist_config.yml", dir_type="corsika")
+    corsika_output_instance.hist_config = output_file
+    assert corsika_output_instance.hist_config == temp_hist_config
+
+
+def test_create_regular_axes_valid_label(corsika_output_instance):
+    hists = ["hist_position", "hist_direction", "hist_time_altitude"]
+    num_of_axes = [3, 2, 2]
+    for i_hist, _ in enumerate(hists):
+        axes = corsika_output_instance._create_regular_axes(hists[i_hist])
+        assert len(axes) == num_of_axes[i_hist]
+        for i_axis in range(num_of_axes[i_hist]):
+            assert isinstance(axes[i_axis], bh.axis.Regular)
+
+
+def test_create_regular_axes_invalid_label(corsika_output_instance):
+    label = "invalid_label"
+    with pytest.raises(ValueError):
+        corsika_output_instance._create_regular_axes(label)
+
+
+def test_create_histograms(corsika_output_instance):
+    # Test once for individual_telescopes True and once false
+    individual_telescopes = [True, False]
+    corsika_output_instance.telescope_indices = [0, 1, 2, 3]
+    num_of_expected_hists = [4, 1]
+    for i_test in range(2):
+        corsika_output_instance._create_histograms(individual_telescopes[i_test])
+        assert corsika_output_instance.num_of_hist == num_of_expected_hists[i_test]
+        assert len(corsika_output_instance.hist_position) == num_of_expected_hists[i_test]
+        assert len(corsika_output_instance.hist_direction) == num_of_expected_hists[i_test]
+        assert len(corsika_output_instance.hist_time_altitude) == num_of_expected_hists[i_test]
+        assert isinstance(corsika_output_instance.hist_position[0], bh.Histogram)
+        assert isinstance(corsika_output_instance.hist_direction[0], bh.Histogram)
+        assert isinstance(corsika_output_instance.hist_time_altitude[0], bh.Histogram)
+
+
+def test_fill_histograms_no_rotation(corsika_output_file_name_string):
+    # Sample test of photons: 1 telescope, 2 photons
+    photons = [
+        {
+            "x": 722.6629,
+            "y": 972.66925,
+            "cx": 0.34438822,
+            "cy": -0.01040812,
+            "time": -115.58354,
+            "zem": 1012681.8,
+            "photons": 2.2303813,
+            "wavelength": -428.27454,
+        },
+        {
+            "x": 983.2037,
+            "y": 809.2618,
+            "cosx": 0.34259367,
+            "cosy": -0.00547006,
+            "time": -113.36441,
+            "zem": 1193760.8,
+            "photons": 2.4614816,
+            "wavelength": -522.7789,
+        },
+    ]
+
+    azimuth_angle = None
```

Comment:
They are used in the `_fill_histograms`. They have to be None because the idea of this test is to test this part of the code, in which no rotations are conducted:
```
            if azimuth_angle is None or zenith_angle is None:
                photon_x, photon_y = photons_info["x"], photons_info["y"]
            else:
                photon_x, photon_y = rotate(
                    photons_info["x"],
                    photons_info["y"],
                    azimuth_angle,
                    zenith_angle,
                )
```

Reply:
Yes but you can simply call `_fill_historgrams` without passing these variables and they will be set to `None` anyway cause that's the default. If you want to be explicit you could call
```
corsika_output_instance_fill._fill_histograms(photons, None, None)
```
or if you want to be super explicit
```
corsika_output_instance_fill._fill_histograms(photons=photons, azimuth_angle=None, zenith_angle=None)
```

Either way you don't need to define these variables here.